flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/runtime/functions/aggfunctions/MaxAggFunctionWithRetract.scala

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.runtime.functions.aggfunctions
+
+import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, TypeInformation, Types}
+import org.apache.flink.table.api.dataview.MapView
+import org.apache.flink.table.dataformat.Decimal
+import org.apache.flink.table.functions.AggregateFunction
+import org.apache.flink.table.runtime.functions.aggfunctions.Ordering._
+import org.apache.flink.table.typeutils.DecimalTypeInfo
+
+import java.lang
+import java.lang.{Iterable => JIterable, Long => JLong}
+import java.sql.{Date, Time, Timestamp}
+
+/** The initial accumulator for Max with retraction aggregate function */
+class MaxWithRetractAccumulator[T] {
```

Comment:
Can we make this Java?

Reply:
ok, should implements MinWithRetractAggFunctionTest as java in this PR?